ios/FluentUI/Pill Button Bar/PillButton.swift

```diff
@@ -147,12 +151,12 @@ open class PillButton: UIButton {
         }
     }
 
-    private let unreadDotLayer: CALayer = {
-        let unreadDotLayer = CALayer()
-        unreadDotLayer.bounds.size = CGSize(width: Constants.unreadDotSize, height: Constants.unreadDotSize)
-        unreadDotLayer.cornerRadius = Constants.unreadDotSize / 2
-        return unreadDotLayer
-    }()
+    private var unreadDotLayer = CALayer()
+
+    private func initUnreadDotLayer() {
```

Comment:
Thanks for the suggestion! I was hoping someone would have a better way to do this :)

Reply:
This new version is not quite what @rdeassis had in mind, I think. Speaking for them for a second (sorry @rdeassis if I get it wrong!), this version will always create and initialize the unread dot's `CALayer`, even if we never use it. Since that's likely to be the case for most users, it'd be nice to have it be both initialized and set up only when it's being used, which Swift allows for with the `lazy` keyword.

But we can't just tag the variable as `lazy`, since it's unconditionally initialized during the button's `init` method (which invokes `initUnreadDotLayer`). The solution @rdeassis suggested would only execute the lambda to create and initialize `unreadDotLayer` the first time it's accessed, which we can delay until when it's being requested for display.